src/fetchService.ts

```diff
@@ -54,77 +51,89 @@ export class FetchService {
 
     // Initialize fetch instances for each content type
     if (config.contentTypes) {
-      Object.entries(config.contentTypes).forEach(([type, settings]) => {
+      for (const [type, settings] of Object.entries(config.contentTypes)) {
         const contentType = type as ContentType;
-        this.fetchInstances.set(contentType, defaults({
-          cachePath: path.join(config.basePath, settings.path),
-          retry: {
-            retries: settings.retries || 3,
-            factor: settings.factor || 2,
-            minTimeout: settings.minTimeout || 1000,
-            maxTimeout: settings.maxTimeout || 5000,
-          },
-          cache: settings.cacheMode || 'default',
-          // Enable ETag and Last-Modified validation
-          cacheAdditionalHeaders: ['etag', 'last-modified']
-        }));
-      });
+
+        // Create a fetch instance with the specified settings
+        this.fetchInstances.set(
+          contentType,
+          /* defaults({
+            cachePath: join(config.basePath, settings.path),
+            retry: {
+              retries: settings.retries || 3,
+              factor: settings.factor || 2,
+              minTimeout: settings.minTimeout || 1000,
+              maxTimeout: settings.maxTimeout || 5000,
+            },
+            cache: settings.cacheMode || 'default',
+            // Enable ETag and Last-Modified validation
+            // cacheAdditionalHeaders: ['etag', 'last-modified'], TODO: see if we can use this, the type definition doesn't allow it but it was defined in the original code
+          }) */
+         fetch
```

Comment:
I have temporarily switched to the global `fetch` module because the `make-fetch-happen` module wasn't playing well with ESM builds - I didn't see any noticeable difference although I suspect some of the caching might now not be happening. This is just an assumption based on what that `make-fetch-happen` package is supposed to be doing.

Reply:
Correct - I wanted to use native fetch but then also wanted caching so `make-fetch-happen` was a quick win at the time at the expense of additional dependencies.  If the plan is to push ahead with llmtext style MD files generated from mkdocs sources then this can all go away which is a win.